File: packages/print-service-client/src/pos-payment-in-ticket.ts
/**
 * Cobro / PAYMENT_IN → agente KaiPrinters (`type: "pos-payment-in-ticket"`).
 */

import type { PosSaleTicketCompany } from "./pos-sale-ticket";

export const POS_PAYMENT_IN_TICKET_PAYLOAD_VERSION = 1;

export type PosPaymentInTicketPaymentRow = {
  label: string;
  amount: number;
  reference?: string | null;
};

export type PosPaymentInTicketAllocationRow = {
  documentNumber: string;
  amount: number;
};

export type PosPaymentInTicketPayload = {
  version: number;
  documentNumber: string;
  issuedAt: string;
  company: PosSaleTicketCompany;
  customerName?: string | null;
  customerDocument?: string | null;
  branchName?: string | null;
  pointOfSaleName?: string | null;
  operatorName?: string | null;
  totalCollected: number;
  amountPaid: number;
  payments: PosPaymentInTicketPaymentRow[];
  allocations: PosPaymentInTicketAllocationRow[];
  notes?: string | null;
  externalReference?: string | null;
};

export type PosPaymentInTicketPrintExtras = {
  filename: string;
  documentType?: string;
  internalFolio?: string;
  sourceApp?: string;
  purpose?: string;
};

File: kai-printers-desktop/src-tauri/src/pos_payment_in_ticket.rs
//! Cobro PAYMENT_IN → JSON (`pos-payment-in-ticket`).

use crate::pos_sale_ticket_pdf::TicketCompany;
use anyhow::{Context, Result};
use serde::Deserialize;
use std::path::PathBuf;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentInLine {
    pub label: String,
    pub amount: f64,
    pub reference: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentInAllocation {
    pub document_number: String,
    pub amount: f64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosPaymentInTicket {
    #[serde(default)]
    #[allow(dead_code)]
    pub version: i32,
    pub document_number: String,
    pub issued_at: String,
    pub company: TicketCompany,
    pub customer_name: Option<String>,
    pub customer_document: Option<String>,
    pub branch_name: Option<String>,
    pub point_of_sale_name: Option<String>,
    pub operator_name: Option<String>,
    pub total_collected: f64,
    pub amount_paid: f64,
    #[serde(default)]
    pub payments: Vec<PaymentInLine>,
    #[serde(default)]
    pub allocations: Vec<PaymentInAllocation>,
    pub notes: Option<String>,
    pub external_reference: Option<String>,
}

pub fn parse_pos_payment_in_ticket_from_value(value: &serde_json::Value) -> Result<PosPaymentInTicket> {
    serde_json::from_value(value.clone()).context("parse pos-payment-in-ticket")
}

pub fn write_pos_payment_in_ticket_escpos_from_value(
    dir: &PathBuf,
    value: &serde_json::Value,
) -> Result<PathBuf> {
    crate::pos_payment_in_ticket_escpos::write_pos_payment_in_ticket_escpos_from_value(dir, value)
}
